src/app/guards/auth.guard.ts

```diff
@@ -13,3 +13,21 @@ export class AuthGuard implements CanActivate {
     return this.authService.getIsUserObservable();
   }
 }
+
+export class LoginGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  canActivate(): boolean {
+    if (
+      this.authService.getIsUserObservable() &&
+      this.authService.getUserData()?.role === 'admin'
+    ) {
+      this.router.navigate(['app-todo-list-component']);
```

Comment:
I tihnk a normal user should be able to see tasks as well and task details, shouldnt have any permissions to edit though